src/components/favourites/favouriteContainer.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import showRepository from '../../repository/tvShowRepository';
+import Spinner from '../ui/spinner';
+import FavouriteList from './favouriteList';
+import { FETCH_STATES } from '../constants';
+
+class FavouriteContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      fetchState: null,
+      shows: [],
+    };
+    this.fetchShows = this.fetchShows.bind(this);
+  }
+
+  componentWillMount() {
+    this.fetchShows(this.props.favourites);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (JSON.stringify(nextProps.favourites) !== JSON.stringify(this.props.favourites)) {
```

Comment:
ogólnie robienie na każdym componentWillReceive... takiego ciężkiego porównania to troche nieoptymalne. Ciężko mi teraz stwierdzić czy jakaś implementacja `deepEqual` byłaby lepsza, ale tutaj odpalany jest stringify 2 razy, więc to też ciężkie.
Jako całkowitą alternatywę moze po prostu traktowanie danych w inny sposób i konkretne porównywanie "z ręki" w zależności od typu danych? obiekty można deepEqualem, array można funkcjami typu `.any()` lub `some()`. Musiałbym się przypatrzyć danym, ale póki co nie robiłbym takiego "globalnego" rozwiązania typu "wszystko do stringa"